Frontend/CO.CDP.OrganisationApp/Pages/Registration/OrganisationIdentification.cshtml.cs

```diff
@@ -1,18 +1,51 @@
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
 
-namespace CO.CDP.OrganisationApp.Pages.SupplierRegistration
+namespace CO.CDP.OrganisationApp.Pages.Registration
 {
     public class OrganisationIdentificationModel : PageModel
     {
         private readonly ILogger<OrganisationIdentificationModel> _logger;
+        public bool HasError { get; set; }
+        private readonly ISession _session;
 
-        public OrganisationIdentificationModel(ILogger<OrganisationIdentificationModel> logger)
+        public OrganisationIdentificationModel(ILogger<OrganisationIdentificationModel> logger, ISession session)
         {
             _logger = logger;
+            _session = session;
         }
 
         public void OnGet()
         {
+            HasError = false;
         }
+
+        public IActionResult OnPost(string organisationIdentification, IDictionary<string, string> organisationNumbers)
```

Comment:
Possibly convert method parameters to Model properties, that way its easy to use inbuild MVC Model Validation framework.

Reply:
Good suggestions, I have done that. Updated and refactored the whole code.